xarray/tests/test_backends.py

```diff
@@ -138,96 +138,110 @@ def open_example_mfdataset(names, *args, **kwargs) -> Dataset:
     )
 
 
-def create_masked_and_scaled_data() -> Dataset:
-    x = np.array([np.nan, np.nan, 10, 10.1, 10.2], dtype=np.float32)
+def create_masked_and_scaled_data(dtype: np.typing.DTypeLike = np.float32) -> Dataset:
```

Comment:
```suggestion
def create_masked_and_scaled_data(dtype: type[np.number] = np.float32) -> Dataset:
```

aha, this is not the standard dtype variable. 

This example passes mypy, so should hopefully work for these functions as well:
```python 
import numpy as np


def test(dtype: type[np.number] = np.float32) -> None:
    print(np.float32(32))

    if isinstance(dtype, type(np.number)):
        print(np.float32(32))


test(np.float32)

```